newmail/lib.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file
+ * New mail notification observer.
+ *
+ * @authors
+ * Copyright (C) 2022 Michal Siedlaczek <michal@siedlaczek.me>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MUTT_NEW_MAIL_H
+#define MUTT_NEW_MAIL_H
+
+#include <stdint.h>
+#include <stdio.h>
+#include "mutt/notify.h"
+#include "format_flags.h"
+
+typedef int(Execute)(const char *cmd);
+
+const char *new_mail_format_str(char *buf, size_t buflen, size_t col, int cols,
+                                char op, const char *src, const char *prec,
+                                const char *if_str, const char *else_str,
+                                intptr_t data, MuttFormatFlags flags);
+int handle_new_mail_event(const char *cmd, struct NotifyCallback *nc, Execute *execute);
+int new_mail_observer(struct NotifyCallback *nc);
```

Comment:
Only `new_mail_observer()` is used from the outside, so we can drop everything else.
Including the `#include`s.  They could be replaced with a forward declaration:
```c
struct NotifyCallback;
```

Reply:
It's actually not entirely true. I am using those functions in tests. I split them to be able to write some unit tests. If I merge all of them together, it will be very difficult to write any tests. I suppose you can hack it and set config to use some specific command that writes to a file and we then read it to verify... Would you prefer I did that?

Reply:
Ah!  I always forget the tests.
Leave them as they are and I'll re-educate myself on their usage :-)

Reply:
Sure; if you see a better way to test these, let me know.